op-node/node/client.go

```diff
@@ -164,3 +169,38 @@ func (cfg *PreparedL1Endpoint) Check() error {
 
 	return nil
 }
+
+type L1BeaconEndpointConfig struct {
+	BeaconAddr string // Address of L1 User Beacon-API endpoint to use (beacon namespace required)
+}
+
+var _ L1BeaconEndpointSetup = (*L1BeaconEndpointConfig)(nil)
+
+func (cfg *L1BeaconEndpointConfig) Setup(ctx context.Context, log log.Logger) (cl client.HTTP, err error) {
+	return client.NewBasicHTTPClient(cfg.BeaconAddr, log), nil
+}
+
+func (cfg *L1BeaconEndpointConfig) Check() error {
+	if cfg.BeaconAddr == "" {
+		return errors.New("expected beacon address, but got none")
+	}
+	return nil
+}
+
+// PreparedL1BeaconEndpoint enables testing with an in-process pre-setup connection to a beacon node mock
+type PreparedL1BeaconEndpoint struct {
```

Comment:
Is this just useful for testing? If so, should it live in `op-e2e`? Or is there prior art in the codebase for having it here?

Reply:
this was for the e2e test which will be another PR, so removed from this one.